### DISCUSSION TITLE
Direction, Vision and Ethics of this Working Group

### DISCUSSION THREAD

**Discussion by mcsadmin** [CredComSoc/ProtocolWorkingGroup#2] [General]
# What we're about here

We would like to try this framework, which looks to a 'hierarchy of scope', where:

-  **Direction** references outcomes we can't directly build, but hope to be able to develop conditions for the emergence of,
-  **Vision** describes the thing which, with ambition, collaboration and coordination of effort resources cross time, we hope to build, and
- **Ethics** set out the constraints on, and requirements to, our action in bring our vision into existence, working to move in that Direction.

Note that the Vision is necessarily imprecisely defined - following the advice of Chris. Alexander's pattern, '[Gradual Stiffening](https://www.iwritewordsgood.com/apl/patterns/apl208.htm)', which suggests that we consider ourselves to be;
`weaving a structure which starts out globally complete, but flimsy; then gradually making it stiffer but still rather flimsy; and only finally making it completely stiff and strong.`

To work toward the vision, we use a **Next Steps** approach, looking to define a small number of things to focus on. Roughly, the thought is the most of the effort should be on immediate blocks to progress, with a small proportion spent attempting to look beyond those immediate tasks, thinking about preparatory work for the further blocks to progress which will be revealed as we progress.

**Comment by JohnG-cc** [comment]
I'm not sure if this is the right thread or if this should be in a different "Scope" thread, but I thought that @utunga raised a good question in the welcome thread ( #1):

> To me it feels to me a big part of the protocol design work we are hopefully doing here will be around creating a wa of meshing together or melding those different ideas togther. (I am especially keen on merging mutual credit and multilateral offset clearing into one model, but also perhaps equity and mutual staking etc could be involved).

>I myself also really wanna explore systems that involve both tokenization and credit and how those two different conceptions of 'what money is' can work together in one. Makes me a bit of an outlier in some parts but yeah. 

Or to paraphrase; Given that @matslats has completed a reference implementation of the Credit Commons Protocol in PHP over on GitLab, to what extent do we want the working group to be focused on the rollout of the existing protocol in software (including any "bugfix" type changes to the protocol itself that are identified), and to what extent do we want the working group to involve consideration of potential extensions, alterations, improvements, or substitutes to/of/for the existing protocol? If both, should we coordinate these separate streams of work into clearly separated sub-projects somehow?

**Comment by matslats** [comment]
To me its sounds as if Utunga is widening the scope too far.
The credit commons is a very particular thing, what I call a 'structure of credit', and it makes little sense to integrate other ideas into it.
The notion of offset clearing exists in a different graph and different context. The graph is what i call a 'mesh credit' structure, where any node is linked to any other node via edges which carry an amount of debt. That is to say it is not a tree. The clearing algorithm identifies circles in the mesh, which it could not do in a tree because in the tree there is only one path possible between any two nodes and all debts along that path are cancelled out. 
Also because offset clearing always looking towards settlement with a limited amount of cash, whereas mutual credit is just cancelling what can be cancelled and settlement if needed happens elsewhere.
You could use the tree to reduce the need for settlement and no algo would be needed. The difference is that the outstanding debts would likely not be with the actual trading partners, which tomaz tells me is a desirable variation in offset clearing, but traders need to give their consent to that and extend some trust to other traders they are not directly linked to.

Another approach is to consider how a mutual credit tree, in its context might bridge to another payment structure like a mesh. In short the mesh can become a node in the tree and the tree can become a node in the mesh. But I don't think that's what Miles was talking about.

Note that arising from the #CoFi gathering something like a 'protocols working group' is planned. This needs to establish standards for mesh credit IMO but I'm concerned it could give undue attention to ECSAs pre-existing and perhaps vague notion of 'Protocols for Post-Capitalist expression. I will be in that space.

**Comment by JohnG-cc** [comment]
I take it then that the purpose of this working group is to be "focused on the rollout of the existing protocol in software (including any "bugfix" type changes to the protocol itself that are identified)", rather than to be considering extending, improving, or designing substitutes for the existing Credit Commons Protocol? (and that any design work like that more properly belongs in the #CoFi protocols group?)

**Comment by matslats** [comment]
That's how I would break things up.
But it's good to remain aware, in this group, of those other possible permutations and use-cases, and I for one intend to have a foot in the other group.

**Comment by utunga** [comment]
I think it is a really good question - to establish the scope that we are interested in working on.

I take the point that perhaps it may be too wide for the ambitions of this discussion, here, if I was thinking that this could be a place to sort out how to merge those different concepts. Don't get me wrong still very excited about that idea but we may or may not have to do that here. 

However, I think I can admit that I'm not all that excited about a protocol working group that is "focused on the rollout of the existing protocol in software, including any 'bugfix' type changes to the protocol itself". That, on the other hand is too narrow for me. Part of this is because I think that the protocol, as currently specified, is hard to scale, and not decentralized enough. 

I hope that it is possible to compromise however in the scope of what we talk about?

I guess we could perhaps we could focus on 'mutual credit' as the main scope (just nailing that) but with inter-group mutual credit (in some sort of fractal self-similar way) as a key part. Definitely want to learn and grow (in a 'bug fix' kind of way) from the experience of using this. 

But yeah, I think that at the very least discussions around incorporating some pretty solid and well known ideas like engineering ideas like idempotent messaging, transport independent, self-signed messages, and looking into the possibility of things like CRDT based/CRDT informed state and even maybe generalizing payment routing (if not payment clearing) should be and could be in scope for this discussion without it turning into a wild 'on a mountain' speculation session a-la the concerns about upcoming co-fi collaboration.

It seems to me that you have already achieved a protocol and an implementation, we have rough working code. Time to get more serious about it... make it scalable, strictly defined and reliable etc. Even if this requires some deeper discussion than has been possible prior to now. 

But of course the scope should be defined by, not just me, but the goals of all the people coming together here. 

@dilgreen would love to hear your thoughts on this also. @JohnG-cc thanks for asking the question. (And of course @matslats it needs to be said thanks so much for all the work getting us to where we are as well as your comments above).

**Reply by dilgreen** [reply]
Thanks for the above.

Coming to clarity on these issues is important.

From thinking and discussing with @utunga , I think this requires another discussion thread on Governance - or, more usefully - ‘How we do things around here’. 

On the basis of clarity there - which I hope will be easy to develop, as there is some strong background - it will be easier to address these scope questions - for which I believe  there is also some useful background which should enable us to move forward here.

Right now, though, it looks like time to head to that [Governance conversation](https://github.com/CredComSoc/ProtocolWorkingGroup/discussions/9).

**Comment by dilgreen** [comment]
Re: upcoming call today.
Google invite has agenda thus:

1. Key is the 'tracks' suggestion - that there is a 'working' track - moving toward a 'Version 1' which is Matt's domain, and a 'Version 2' track, which is learning from Version 1 and also potentially including significant changes. 
2. But it will be useful, before that,  to have a timeboxed conversation as to look at simple statements of 'Direction' and 'Vision' (without needing to nail these down)
3. ![DirectionVisionSteps-ethics cartoon](https://github.com/CredComSoc/ProtocolWorkingGroup/assets/5934679/dc1fb16e-0338-41ae-b833-30846d816794)
4. Credit Commons society have asked this group to think about licensing - what does it mean to have CredCom Open Source?

Hopefully appropriately, I thought it might be useful to suggest a start point for Direction and Vision. In the sociocratic approach, decisions typically come with  'review date' - in other words, _we consented to policy <x>, but we'll review it after <a time period>._

Since these statements are, by intent, quite serious, and since we haven't discussed together before, any idea that we'll arrive at definitive statements in a time-boxed period today seems wrong-headed.

What might be possible, I'm thinking, is to use the time to look at these as 'stick man' versions of what we might end up deciding upon, that we can do a quick and dirty edit  on, and then consent to on the basis of a review in a couple of months or so.

Obviously, if the mood is to do something else, we'll do something else.

So, here goes;

### Stick-man Direction statement:
(a North Star: unattainable, but always available as a guide through confusion or controversy)

**Readily available, high quality software tools for sophisticated, non-extractive economic coordination at scale through unitised promise accounting, designed specifically to support social-scale productive capacity**

### Stick-man Vision statement:
(An ambitious statement about the wonderful thing we believe we just might be able to build, for real)

**A Credit Commons Protocol software suite, capable of being used by people of all kinds, to establish economic contexts under their own control, which can operate recursively, so that groups of Credit Commons instances can use the same software to form a federation tree.**
This software will be (in no particular order):
   - of the highest achievable quality
   - built through self-governing collaboration, as a Commons, with reference to Ostrom Principles
   - free to instantiate on the terms of an appropriate Commons license (which may not yet exist)
   - well documented
   - easy to instantiate (on a par with wordpress)
   - carefully and rigorously scoped and structured in order to offer excellent adaptability and compostability
   - always developed in accordance with the most positive interpretation of the agreed Direction achievable.

@utunga @matslats @JohnG-cc @tfwoodroofMCS

**Reply by dilgreen** [reply]
[editable gdoc for notes and Direction/Vision etc](https://docs.google.com/document/d/1Ro9GFzNYA562pUiaBOBXupQuIRswf_he7vjBJeA7L3Y/edit?usp=sharing)
